src/auth_db/mod.rs

```diff
@@ -8,23 +8,73 @@ use std::{
 
 use hex;
 use reqwest::{Client as RequestClient, Error as RequestError, StatusCode, Url, UrlError};
+use serde::de::{Deserialize, Deserializer, Error as DeserializeError, Unexpected};
 
 use settings::Settings;
 
 #[cfg(test)]
 mod test;
 
-#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq)]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub enum BounceType {
-    Hard = 1,
-    Soft = 2,
-    Complaint = 3,
+    Hard,
+    Soft,
+    Complaint,
+}
+
+impl<'d> Deserialize<'d> for BounceType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'d>,
+    {
+        let value: u8 = Deserialize::deserialize(deserializer)?;
+        match value {
+            0 => {
+                // Set whenever the auth db receives a value it doesn't recognise
+                println!("Mapped default auth db bounce type BounceType::Soft");
+                Ok(BounceType::Soft)
```

Comment:
This match arm is the reason why I couldn't derive `Deserialize` any more. I wanted to ensure we had sane behaviour if we read any zeroes from the db and the safest thing to do seemed like treat it as a soft bounce but log when we're doing so.

Note that this differs from the behaviour in the auth server (see mozilla/fxa-auth-server#2453).